File: examples/governance_static_go/domain/suspend_model.go
package domain

import (
	"time"
	"fmt"
)

type SuspendModel struct {
	ModelId string `json:"model_id"`
}

func (c SuspendModel) CommandName() string { return "SuspendModel" }

func (c SuspendModel) Execute(repo AiModelRepository) (*AiModel, *SuspendedModel, error) {
	existing, err := repo.Find(c.ModelId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("AiModel not found: %s", c.ModelId)
	}
	if existing.Status != "approved" && existing.Status != "classified" && existing.Status != "draft" {
		return nil, nil, fmt.Errorf("cannot SuspendModel: AiModel is in %s state", existing.Status)
	}
	existing.Status = "suspended"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := SuspendedModel{
		AggregateID: existing.ID,
		ModelId: c.ModelId,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/model_registry_domain/lib/model_registry_domain/ai_model/commands/suspend_model.rb
module ModelRegistryDomain
  class AiModel
    module Commands
      class SuspendModel
        include Hecks::Command
        emits "SuspendedModel"

        attr_reader :model_id

        def initialize(model_id: nil)
          @model_id = model_id
        end

        def call
          existing = repository.find(model_id)
          if existing
            unless ["approved", "classified", "draft"].include?(existing.status)
              raise ModelRegistryDomain::Error, "Cannot SuspendModel: status must be one of approved, classified, draft, got '#{existing.status}'"
            end
            AiModel.new(
              id: existing.id,
              name: existing.name,
              version: existing.version,
              provider_id: existing.provider_id,
              description: existing.description,
              risk_level: existing.risk_level,
              registered_at: existing.registered_at,
              parent_model_id: existing.parent_model_id,
              derivation_type: existing.derivation_type,
              capabilities: existing.capabilities,
              intended_uses: existing.intended_uses,
              status: "suspended"
            )
          else
            raise ModelRegistryDomain::Error, "AiModel not found: #{model_id}"
          end
        end
      end
    end
  end
end
